src/trace/tracer.ts

```diff
@@ -147,24 +154,24 @@ class ContextTracer implements SnapshotterDelegate, ActionListener {
     return (page as any)[pageIdSymbol];
   }
 
-  async onActionCheckpoint(name: string, metadata: ActionMetadata): Promise<void> {
+  async onActionCheckpoint(name: string, sdkObject: SdkObject, metadata: CallMetadata): Promise<void> {
     const snapshotId = createGuid();
     snapshotsForMetadata(metadata).push({ name, snapshotId });
-    await this._snapshotter.forceSnapshot(metadata.page, snapshotId);
+    await this._snapshotter.forceSnapshot(sdkObject.attribution.page!, snapshotId);
```

Comment:
I _think_ we are coming here for all actions now, so `page` could easily be null, e.g. `context.newPage()`.